docker/build-tools/Dockerfile

```diff
@@ -240,6 +245,32 @@ RUN gem install --no-wrappers --no-document mdl -v ${MDL_VERSION}
 RUN gem install --no-wrappers --no-document html-proofer -v ${HTMLPROOFER_VERSION}
 RUN gem install --no-wrappers --no-document awesome_bot -v ${AWESOMEBOT_VERSION}
 
+#############
+# Python
+#############
+
+FROM ubuntu:xenial as python_context
+
+ENV YAMLLINT_VERSION=1.17.0
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    ca-certificates \
+    curl \
+    libc-dev \
+    pkg-config \
+    python3
+
+# Install pip
+RUN curl "https://bootstrap.pypa.io/get-pip.py" -o "get-pip.py"
+RUN python3 get-pip.py --no-wheel
+
+# Install yamllint
+RUN pip install yamllint==${YAMLLINT_VERSION}
+
+# Clean up stuff we don't need in the final image
+RUN rm -rf /usr/local/bin/pip*
+RUN rm -fr /usr/local/lib/python3.5/dist-packages/pip
+
```

Comment:
Another minor nit would be adding the required list for python similar to the others (golang, general, ruby).